### DISCUSSION TITLE
[Suggestion] Add folder content preview in nemo file manager

### DISCUSSION THREAD

**Discussion by calyrex-s** [linuxmint/discussions#388] [Ideas]
Nemo is like the perfect file manager but it still needs the option to preview images and videos in the folder icon. Just like Dolphin, Windows explorer,..

![Screenshot 2024-04-12 104241](https://github.com/linuxmint/discussions/assets/166698385/db9fb1e8-09f3-45c8-b328-941b029911fc)

**Comment by BlabberBlabbing** [comment]
Completely second this. I'm considering other desktops exactly because of this missing feature of Cinnamon.

Furthermore you forget mentioning what IMO is one of the main uses for a file explorer "well behaved embedded preview" feature (for lack of a standard def). 
The preview using the backspace button (to preview and close), while good enough for single media, is uncomfortable for multiple files - for example fastly scrolling through an image folder - with or without thumbnail view. 
Also if you want to resize the preview, it looks like you have to make it manually or you might have to fiddle with configuration files: in both cases an embedded preview could allow a more flexible and easy way to customize sizing.

You already have implemented the "extra pane", is it that hard to have an option to automatically load there (resized to fit the window) the preview, otherwise triggered manually by the backspace?

Thanks

![image](https://github.com/user-attachments/assets/bfcfae1c-3677-4324-b515-876a2eddc385)

**Comment by Jopp-gh** [comment]
Linux in general tries to cut the **** and is unlike commercial bloat /spy/virusware.
 
On the other hand, Linux is very customizable, one aspect which requires engagement and help yourself- or do it yourself abilities. 

Btw, users don't have to reinvent software/features they want, but check online for suitable plugins. And... add their own code modifications if possible. That's how open source works. People who help each other. 

Here's a great resource:
https://github.com/smurphos/nemo_actions_and_cinnamon_scripts

Don't forget to give them a star🌟

**Reply by BlabberBlabbing** [reply]
I guess you are answering my post as well since you refer to "reinventing yourself" and posted just after I did.

Unfortunately, while I'm not sure these "actions" scripts solve OP's problem, I'm confident they don't solve mine (which might be just my mistake for misinterpreting what the OP asked for, and posting here). I might just open a dedicatd post later.

**Reply by Jopp-gh** [reply]
I see. Other quick solutions are:  
[Allusion](https://allusion-app.github.io/)
[Sigma file manager](https://github.com/aleksey-hoffman/sigma-file-manager)

I'm more than happy with the scripts suggested in my post before, so there's nothing more I would suggest, apart modifying scripts or creating a plugin by ourself. 
There might be other plugins online doing a better job, but I'm not aware of them.

**Reply by BlabberBlabbing** [reply]
Allusion is an image library manager.
I don't see how, to fix a file manager missing feature, you find reasonable to bloat your OS with something you don't need - but I guess I'm an "order maniac"...

For Sigma file manager: it looks cool, but I have already wasted many hours to test/customize a well known, stable FM like Nemo on Cinnamon.
Going the route of alpha/beta testing new SW is a nono.

According to ChatGPT the only stable, full featured one that has the side automatic preview pane is Dolphin. Which supposedly "brings KDE dependencies which might feel out of place": I don't know exactly what this means but it doesn't sound good and I don't feel like testing it.

I guess I'll go [here](https://github.com/linuxmint/nemo)

**Reply by Jopp-gh** [reply]
>bloat your OS with something you don't need

Allusion isn't "bloat" - its a great way to inspect quickly a large amount of images. 

Preview thumbnails are best for images, especially if you deal with as lot of images everyday. Not all users have the same needs. You cant read books from thumbnails, just to add some obvious/silly example, therefore suggesting an app to preview images isn't pointless at all. 

Bloat is stuff users can't use, inspect or remove but it bundled with commercial software.

Note, I also added "quick solution". Your aren't forced to use either what I'm suggesting.

**Reply by BlabberBlabbing** [reply]
Bloat to me is just filling your system with unnecessary stuff that may (or may not) be detrimental to your operations.

A full-featured file manager to me is a good thing.
Installing an application that does 100 things, included the one thing you need because your file manager doesn't do it, to me is bloat.

I appreciate your suggestions, but unfortunately what I'm looking for is a good file manager.

**Comment by ghost** [comment]
Bumping this thread, I also need this feature, and this is one major feature I'm missing that KDE's Dolphin provide. Above comments are way too hacky, which I don't want to do with my system, to bork it somehow... I need to know that my system's integrity is intact. So we need this feature to be implemented by the official MINT team, to get quality content, out of the box.

Nemo folder preview has to be implemented, in such a fashion, that other users who find it clutter, can toggle it off, but those who need it can toggle it on.

**Reply by cplittlfield** [reply]
> I also need this feature, and this is one major feature I'm missing that KDE's Dolphin provide[s]. 

And yet, for some reason, you switched from Kubuntu or whatever to Linux Mint, despite having to have that feature.  So it would seem the lesson to derive from this situation is that there is no perfect distro, and the only way to get what we truly want is to patch the source code ourselves.

My attitude to seeing previews inside folder icons, back when I was using the OS I refuse to name, was lukewarm.  After all, I had folders with hundreds of images in them, and there was no way to get more than four to show up in the folder icon.   Neither was there any way to control which four images got to be in the preview.

In the time since I switched to Linux Mint, I found I missed this feature a bit at first, but as I got used to all the features Linux Mint does provide, I stopped missing it.

On the other hand, it would definitely be a useful feature to automatically show a preview of the selected file in a panel to the right of the file window.  The GTK file chooser does this, and so does Pix.  I corresponded for a while with a user who owned a ranch and needed an easier way to see previews of her horse photos.  I might not use such a feature myself, but this woman made me see that there is a strong business use case to be made for such a feature.

**Reply by ghost** [reply]
> And yet, for some reason, you switched from Kubuntu or whatever to Linux Mint

Hehe, no it was actually opensuse, debian and endeavouros where I got used to KDE.

> So it would seem the lesson to derive from this situation is that there is no perfect distro, and the only way to get what we truly want is to patch the source code ourselves.

Yepp that is what we are doing, sharing our experiences to iron into what is already good, but to make it even better, to make a distro that universally fits for everyone's needs.

By the way, yeah — the way preview works in KDE's Dolphin is magical when there are lots of media files in a folder. It works in such a way that, when you hover your mouse over the folder, it dynamically rotates the thumbnails of its contents. This makes it easier to get a better visual understanding of what's inside.

Honestly if Nemo would be as feature rich as Dolphin is, I'd be the happiest person on the earth.

**Comment by BlabberBlabbing** [comment]
For the historians: I've just tested Dolphin and its file preview doesn't really offer what I have in Windows.
While it has a feature I had never seen (maybe I'll grow fond of it) which is the "hover preview" which previews any file you're hovering your mouse on, it lacks document navigation. Dolphin previews the first page of your txt, odt or pdf, but you can't navigate them as happens since forever in windows (screenshot).
![image](https://github.com/user-attachments/assets/cec8c391-b03e-4982-b50c-900a99d98dfb)
Which is odd: you can navigate movies and music, but you can't navigate a txt...
Still Nemo missing these features is really pushing me to work on developing it...
